torch/fx/experimental/proxy_tensor.py

```diff
@@ -933,6 +1026,12 @@ def make_fx(f,
     if decomposition_table is None:
         decomposition_table = {}
 
+    if torch.ops.aten.sym_numel.default not in decomposition_table:
+        decomposition_table = {
+            **decomposition_table,
+            torch.ops.aten.sym_numel.default: torch._decomp.decompositions.sym_numel
+        }
```

Comment:
What's going on here?

Reply:
I want the sym_numels to be expressed in terms of inputs. This is reduces uses on tensors, but is also needed because I had a scenario where a SymNumel was first used in a tensor in the backward that ends up int he backward and then later used in a tensor that is in the forward. 

Is there a better way of decomposing sym_numel? I tried adding some logic [https://github.com/pytorch/pytorch/pull/116158/files#diff-6960ce90e7162c0953b1ca07e92e7f0f2f6ba63b427b42df593e20cc6a096bb7R254](here) but it added detach nodes into the graph for reasons I couldnt understand.